snap/snapcraft.yaml

```diff
@@ -274,7 +279,11 @@ parts:
       export CGO_LDFLAGS_ALLOW="(-Wl,-wrap,pthread_create)|(-Wl,-z,now)"
 
       # Check that `golangci-lint` is happy with the code.
-      golangci-lint run --verbose
+      if command -v golangci-lint; then
```

Comment:
nit: We could use more explicit condition here based on [`CRAFT_ARCH_BUILD_ON `](https://snapcraft.io/docs/parts-environment-variables#heading--snapcraft-configuration-core24)

Reply:
That is indeed an option, I chose not to do that as we have to declare an assertion on architecture in the `build-snaps` already to avoid build failure on the back of the snap not being available:
```yaml
      - to riscv64:
        - go/1.22/stable
      - else:
        - go/1.22/stable
        - golangci-lint
```

With that in mind it felt a bit too much to repeat that assertion here.